zmq_addon.hpp

```diff
@@ -675,20 +675,24 @@ class active_poller_t
     {
         if (!handler)
             throw std::invalid_argument("null handler in active_poller_t::add");
-        auto it = decltype(handlers)::iterator{};
-        auto inserted = bool{};
-        std::tie(it, inserted) = handlers.emplace(
-          socket, std::make_shared<handler_type>(std::move(handler)));
+        auto it = handlers.find(socket);
+        if (it != handlers.end()) {
+            // add to poller to make in throw the expected error code
+            base_poller.add(socket, events, nullptr);
```

Comment:
I don't completely understand the comment and why `base_poller.add` is called. Why can't we throw the expected exception directly from here?

Reply:
Good question. What I would like to do is set errno and then throw error_t. But Im not sure what is the best platform independent way to set errno is, similar to zmq_errno.

This is so so that double add throws the same kind of exception as in poller_t, but that may not be required if too hard to implement properly.

Reply:
If the sole purpose is to get the right errno into `error_t` why not add a constructor to `error_t` that accepts an error number? We don't need to set `errno` at all.